mppb/parser.go

```diff
@@ -37,280 +43,206 @@ var headersMap = []map[string]int{
-	for _, f := range files {
-		if dataType(f) == INDENIZACOES {
+	mapIndenizacoes := map[string][]string{}
+	const INDENIZACOES_MATRICULA = 0
+	for _, f := range indenizacoes {
+		mapIndenizacoes[f[INDENIZACOES_MATRICULA]] = f
+	}
+	for _, f := range arquivos {
+		if tipoCSV(f) == INDENIZACOES {
 			continue
 		}
-
-		data, err := dataAsSlices(f)
+		dados, err := dadosParaMatriz(f)
 		if err != nil {
-			return nil, nil, fmt.Errorf("error trying to parse data as slices(%s): %q", f, err)
+			return nil, fmt.Errorf("erro na tentativa de transformar os dados em matriz (%s): %q", f, err)
 		}
-		if len(data) == 0 {
-			return nil, nil, fmt.Errorf("No data to be parsed. (%s)", f)
+		if len(dados) == 0 {
+			return nil, fmt.Errorf("Não há dados para serem parseados. (%s)", f)
 		}
-
-		remu, contra_cheque, ok := retrieveEmployees(data, perks, chave_coleta, f)
+		contra_cheque, ok := getMembros(dados, mapIndenizacoes, chave_coleta, f)
 		if !ok {
 			parseErr = true
 		}
 		folha = append(folha, contra_cheque...)
-		remuneracoes = append(remuneracoes, remu...)
 	}
 	if parseErr {
-		return &FolhaDePagamento{ContraCheque: folha}, &Remuneracoes{Remuneracao: remuneracoes}, fmt.Errorf("parse error")
+		return &proto.FolhaDePagamento{ContraCheque: folha}, fmt.Errorf("parse error")
 	}
-	return &FolhaDePagamento{ContraCheque: folha}, &Remuneracoes{Remuneracao: remuneracoes}, nil
+	return &proto.FolhaDePagamento{ContraCheque: folha}, nil
 }
 
-func retrievePerksData(files []string) ([][]string, error) {
+// getDadosIndenizacoes retorna a planilha de indenizações em forma de matriz
+func getDadosIndenizacoes(files []string) ([][]string, error) {
 	for _, f := range files {
-		if dataType(f) == INDENIZACOES {
-			return dataAsSlices(f)
+		if tipoCSV(f) == INDENIZACOES {
+			return dadosParaMatriz(f)
 		}
 	}
 	return nil, nil
 }
 
-func retrieveEmployees(emps [][]string, perks [][]string, chave_coleta string, fileName string) ([]*Remuneracao, []*ContraCheque, bool) {
+// getMembros retorna o array com a folha de pagamento da coleta.
+func getMembros(membros [][]string, mapIndenizacoes map[string][]string, chaveColeta string, fileName string) ([]*proto.ContraCheque, bool) {
 	ok := true
-	var remuneracoes []*Remuneracao
+	var contraCheque []*proto.ContraCheque
 	counter := 1
-	var contra_cheque []*ContraCheque
-	for _, emp := range emps {
+	for _, membro := range membros {
 		var err error
-		var newEmp *ContraCheque
-		var newRem []*Remuneracao
-		empPerks := retrievePerksLine(emp[0], perks)
-		if newEmp, newRem, err = newEmployee(emp, empPerks, chave_coleta, counter, fileName); err != nil {
+		var novoMembro *proto.ContraCheque
+		indenizacoesMembro := getIndenizacaoMembro(membro[0], mapIndenizacoes)
+		if novoMembro, err = criaMembro(membro, indenizacoesMembro, chaveColeta, counter, fileName); err != nil {
 			ok = false
-			logError("error retrieving employee from %s: %q", fileName, err)
+			logError("error na criação de um novo membro %s: %q", fileName, err)
```

Comment:
Se o log error não for utilizado nenhuma vez, remover e substituir por um log. Fatal.

Acho que com isso removemos de vez o arquivo `util.go` 🥳